### DISCUSSION TITLE
Unable to update applciation config and new release is stuck at the loading page

### DISCUSSION THREAD

**Discussion by tomg-formpak** [awslabs/aws-saas-boost#294] [Q&A] [answered]
Hi Team

I am getting the below two issues. Any support from the team will be great. 

1. Unable to update the application configuration in the application tab - domain name and SSL certificate. It's showing the error,  "Unable to update application configuration". But I can update the service details **(v2.0.0-rc1)**. **Why is this error showing?** 

I tried deleting the SaaS Boost deployment from installer options ( also CloudFormation stacks and S3 buckets) for a fresh installation, but on installing again previous application configurations are auto-filled (prefilled) in the admin panel (which I am not able to update/modify). Please find the screenshot for the error. **Why the configurations from the previous installation are showing in the admin dashboard even after deleting CF stacks and S3 buckets? Do I need to delete or do something else for a fresh installation?** 

![WhatsApp Image 2022-09-18 at 11 35 28 PM](https://user-images.githubusercontent.com/99970882/190931978-0eaffc15-af30-4b8b-9c27-57a82ce0d23b.jpeg)

2. Also, on trying with **v2.0.0-rc2**, the admin dashboard is not opening, it's stuck in the loading page on login (CORS error in the API). Please see the below screenshot for the error. 

<img width="1088" alt="save" src="https://user-images.githubusercontent.com/99970882/190931981-74bcd7a9-94a8-4d48-8960-f5874b48c806.png">

Please advise me on the above two issues. @PoeppingT @brtrvn 

Thank you

**Comment by PoeppingT** [comment, marked as answer]
@tomg-formpak happy to help.

In both cases your issues are likely because your executed delete was unsuccessful. Can you clarify which version you used to delete? `v2.0.0-rc2` contains an all-around fix for delete (https://github.com/awslabs/aws-saas-boost/commit/f6473b4b668a3aa3cb5a53c8f4e0a026745669e5), and you aren't required to use the same version to delete as you did to create.

To keep moving forward I would recommend you
1. Download `v2.0.0-rc2`
2. Use that to delete your environment.
3. Double-check Systems Manager, S3, and CloudFormation to make sure there are no leftover resources, deleting them if they are (this should no longer be necessary after installing/deleting using `v2.0.0-rc2` and above)
4. Install your new environment using `v2.0.0-rc2`.

More context on the above follows:
1. Regarding form items being prefilled from previous installations: in SaaS Boost the application parameters are hosted in the SSM ParameterStore. If previous environment's configurations are showing up in the Admin UI, that means that the Settings service is finding old parameters in the ParameterStore. After running a delete using the install script, try checking Systems Manager in the AWS Console for any lingering parameters and deleting those. Note that this should no longer happen if deleting using v2.0.0-rc2 (and please open a bug in GitHub if you find it).
2. Regarding the "unable to update application configuration" and "CORS" error, both are likely due to failed Settings service calls. The "unable to update application configuration" error will happen if the Settings service throws some exception when a call to `update-config` is made, and the CORS error indicates a failure response from API Gateway (likely again due to Settings service failure responses). To check this you can look at CloudWatch in the AWS Console, click on LogGroups on the left, and look for `sb-${env}-settings-get-config` and `sb-${env}-settings-update-config`. Errors there will show you what's wrong.

Thanks for reaching out, and we appreciate the help kicking the tires on this code! Please keep us updated on your progress and let us know if you hit any other issues.

**Reply by tomg-formpak** [reply]
Thank you @PoeppingT It's working. I will update you with the progress or on any issues.
